runtime/src/main/java/org/corfudb/security/tls/TlsUtils.java

```diff
@@ -4,134 +4,102 @@
-import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.SSLException;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * Utilities for common options parsing and session configuration for
  * encrypted & authenticated TLS sessions.
  */
-
+@Slf4j
 public class TlsUtils {
-    public enum SslContextType { SERVER_CONTEXT, CLIENT_CONTEXT }
-
-    /**
-     * Create SslContext object based on Getopt-style parameter spec.
-     *
-     * @param desiredType Server or client context
-     * @param opts Getopt-style parameters
-     * @param keyStoreException Consumer for key store error
-     * @param ksPasswordFileException Consumer for ks password file error
-     * @param trustStoreException Consumer for trust store error
-     * @param tsPasswordFileException Consumer for ts password file error
-     * @return SslContext object or null on error
-     */
-    public static SslContext enableTls(SslContextType desiredType,
-                                       Map<String, Object> opts,
-                                       Consumer<Exception> keyStoreException,
-                                       Consumer<Exception> ksPasswordFileException,
-                                       Consumer<Exception> trustStoreException,
-                                       Consumer<Exception> tsPasswordFileException) {
-        return enableTls(desiredType,
-                (String) opts.get("--keystore"), keyStoreException,
-                (String) opts.get("--keystore-password-file"), ksPasswordFileException,
-                (String) opts.get("--truststore"), trustStoreException,
-                (String) opts.get("--truststore-password-file"), tsPasswordFileException);
-    }
-
     /**
      * Create SslContext object based on a spec of individual configuration strings.
      *
-     * @param desiredType Server or client context
-     * @param keyStore Key store path string
-     * @param keyStoreException Consumer for key store error
+     * @param isServer Server or client
+     * @param keyStorePath Key store path string
      * @param ksPasswordFile Key store password file string
-     * @param ksPasswordFileException Consumer for ks password file error
-     * @param trustStore Trust store path string
-     * @param trustStoreException Consumer for trust store error
+     * @param trustStorePath Trust store path string
      * @param tsPasswordFile Trust store password file path string
-     * @param tsPasswordFileException Consumer for ts password file error
      * @return SslContext object or null on error
+     * @throws SSLException
+     *          Thrown when there's an issue with loading the trust store.
      */
-    public static SslContext enableTls(SslContextType desiredType,
-                                       String keyStore,
-                                       Consumer<Exception> keyStoreException,
+    public static SslContext enableTls(boolean isServer,
+                                       String keyStorePath,
```

Comment:
You can mark these parameters with `@NonNull` to avoid null checks.

Reply:
Added.